lib/src/main/java/com/diffplug/spotless/FileSignature.java

```diff
@@ -38,70 +38,66 @@
 	 * incremental builds.
 	 */
 	@SuppressFBWarnings("SE_TRANSIENT_FIELD_NOT_RESTORED")
-	private final transient List<File> sortedFiles;
+	private final transient Collection<File> files;
 
-	@SuppressWarnings("unused")
 	private final String[] filenames;
-	@SuppressWarnings("unused")
 	private final long[] filesizes;
-	@SuppressWarnings("unused")
 	private final long[] lastModified;
 
-	public static FileSignature from(File... files) throws IOException {
-		return from(Arrays.asList(files));
+	/** Creates file signature whereas order of the files remains unchanged. */
+	public static FileSignature fromList(File... files) throws IOException {
+		return fromList(Arrays.asList(files));
 	}
 
-	public static FileSignature from(Iterable<File> files) throws IOException {
-		List<File> sortedFiles = toSortedSet(Objects.requireNonNull(files));
+	/** Creates file signature whereas order of the files remains unchanged. */
+	public static FileSignature fromList(Collection<File> files) throws IOException {
+		return new FileSignature(files);
+	}
+
+	/** Creates file signature insensitive to the order of the files. */
+	public static FileSignature fromSet(File... files) throws IOException {
+		return fromSet(Arrays.asList(files));
+	}
+
+	/** Creates file signature insensitive to the order of the files. */
+	public static FileSignature fromSet(Collection<File> files) throws IOException {
+		return new FileSignature(toSortedSet(Objects.requireNonNull(files)));
+	}
+
+	private FileSignature(final Collection<File> files) throws IOException {
+		this.files = files;
 
-		String[] filenames = new String[sortedFiles.size()];
-		long[] filesizes = new long[sortedFiles.size()];
-		long[] lastModified = new long[sortedFiles.size()];
+		filenames = new String[this.files.size()];
+		filesizes = new long[this.files.size()];
+		lastModified = new long[this.files.size()];
```

Comment:
Just wanted to emphasize that we do not store new information here, but only processing results from other information stored by `this` instance.
Ok for you?